addons/website/static/src/builder/plugins/options/dynamic_snippet_option_plugin.js

```diff
@@ -457,5 +460,29 @@ export function setDatasetIfUndefined(snippetEl, optionName, value) {
         snippetEl.dataset[optionName] = value;
     }
 }
+export class SectionTitleAction extends BuilderAction {
+    static id = "sectionTitle";
+    static dependencies = ["dynamicSnippetOption"];
+    apply({ editingElement: el, params: { mainParam: position } }) {
+        // Adjust dataset if title is aside and content-width is not max.
+        if (el.closest(".container-fluid")) {
+            return;
+        }
+        const parentEl = el.closest(".s_dynamic_snippet_products");
+        parentEl.dataset.numberOfElements = position === "left" ? 2 : 4;
+    }
+}
 
+patch(SetContainerWidthAction.prototype, {
+    apply({ editingElement }) {
+        super.apply(...arguments);
+        if (editingElement.classList.contains("s_dynamic_snippet_container")) {
+            const titleEl = editingElement.querySelector(".s_dynamic_snippet_title");
+            const isContentWidthMax = editingElement.classList.contains("container-fluid");
+            const isTitleAside = titleEl?.classList.contains("s_dynamic_snippet_title_aside");
+            const parentEl = editingElement.parentElement;
+            parentEl.dataset.numberOfElements = isContentWidthMax ? 4 : isTitleAside ? 2 : 4;
+        }
+    },
```

Comment:
This could be simplified like this : 
```Suggestion
     apply({ editingElement, params: { mainParam: className } }) {
        super.apply(...arguments);
        if (editingElement.classList.contains("s_dynamic_snippet_container")) {
            const isContentWidthMax = className === "container-fluid";
            const titleAsideEl = editingElement.querySelector(".s_dynamic_snippet_title_aside");
            const parentEl = editingElement.parentElement;
            parentEl.dataset.numberOfElements = !isContentWidthMax && titleAsideEl ? 2 : 4;
        }
    },
```